### ISSUE TITLE
v0.6.0 ClusterRoleBinding with wrong namespace

### ISSUE THREAD

**Issue by blezoray** [kube-green/kube-green#490]
Hello,

I just tested to upgrade Kube-Green on my Openshift cluster from 0.5.2 to 0.6.0 with OLM (Operator Lifecyle Manager).

The pod reports these error message:
```
W1115 08:32:05.804654 1 reflector.go:561] pkg/mod/k8s.io/client-go@v0.31.2/tools/cache/reflector.go:243: failed to list *v1alpha1.SleepInfo: sleepinfos.kube-green.com is forbidden: User "system:serviceaccount:mpms-kubegreen-operator:kube-green-controller-manager" cannot list resource "sleepinfos" in API group "kube-green.com" at the cluster scope
E1115 08:32:05.804703 1 reflector.go:158] "Unhandled Error" err="pkg/mod/k8s.io/client-go@v0.31.2/tools/cache/reflector.go:243: Failed to watch *v1alpha1.SleepInfo: failed to list *v1alpha1.SleepInfo: sleepinfos.kube-green.com is forbidden: User \"system:serviceaccount:mpms-kubegreen-operator:kube-green-controller-manager\" cannot list resource \"sleepinfos\" in API group \"kube-green.com\" at the cluster scope" logger="UnhandledError"
W1115 08:32:06.711916 1 reflector.go:561] pkg/mod/k8s.io/client-go@v0.31.2/tools/cache/reflector.go:243: failed to list *v1alpha1.SleepInfo: sleepinfos.kube-green.com is forbidden: User "system:serviceaccount:mpms-kubegreen-operator:kube-green-controller-manager" cannot list resource "sleepinfos" in API group "kube-green.com" at the cluster scope
E1115 08:32:06.711968 1 reflector.go:158] "Unhandled Error" err="pkg/mod/k8s.io/client-go@v0.31.2/tools/cache/reflector.go:243: Failed to watch *v1alpha1.SleepInfo: failed to list *v1alpha1.SleepInfo: sleepinfos.kube-green.com is forbidden: User \"system:serviceaccount:mpms-kubegreen-operator:kube-green-controller-manager\" cannot list resource \"sleepinfos\" in API group \"kube-green.com\" at the cluster scope" logger="UnhandledError"
W1115 08:32:08.541242 1 reflector.go:561] pkg/mod/k8s.io/client-go@v0.31.2/tools/cache/reflector.go:243: failed to list *v1alpha1.SleepInfo: sleepinfos.kube-green.com is forbidden: User "system:serviceaccount:mpms-kubegreen-operator:kube-green-controller-manager" cannot list resource "sleepinfos" in API group "kube-green.com" at the cluster scope
E1115 08:32:08.541282 1 reflector.go:158] "Unhandled Error" err="pkg/mod/k8s.io/client-go@v0.31.2/tools/cache/reflector.go:243: Failed to watch *v1alpha1.SleepInfo: failed to list *v1alpha1.SleepInfo: sleepinfos.kube-green.com is forbidden: User \"system:serviceaccount:mpms-kubegreen-operator:kube-green-controller-manager\" cannot list resource \"sleepinfos\" in API group \"kube-green.com\" at the cluster scope" logger="UnhandledError"
```

The reason is that the `ClusterRoleBinding` has the namespace of the ServiceAccount which is set to `operators` and not to the local namespace where the operator is deployed.

```yaml
kind: ClusterRoleBinding
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: kube-green-manager-rolebinding
subjects:
  - kind: ServiceAccount
    name: kube-green-controller-manager
    namespace: operators
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: kube-green-manager-role
```
For me, this [CRB](https://github.com/kube-green/kube-green/blob/main/bundle/manifests/kube-green-aggregate-manager-role_rbac.authorization.k8s.io_v1_clusterrole.yaml) should not be part of the manifest and ClusterServiceVersion.

Rgds, Bruno.

**Comment by davidebianchi** [comment]
Hi! I got it, the CRB is needed to give to the operator the correct permission on the resources, but olm does not support aggregator  role. I will release a patch of this as soon as possible!

**Comment by davidebianchi** [comment]
I went deeper on the issue, but it seems that with olm there is not a way to support aggregator role with cluster role (and cluster role binding). This is due to a limitation in the name of the namespace, and it seems there is not a way to set a template for the namespace name.

Due to this limitation, I'd add on the documentation that with olm it is only possible to install inside the `operators` namespace (which is the default one suggested by operatorhub). If you need to install in another namespace, I would suggest to use instead the helm chart.

The other possibilities I thought:
- avoid to use the aggregator role in olm and give to kube-green the permissions only on the default resources. This would permit to install in any namespace, but the main feature of the 0.6 (handle all resource kinds) will be unusable
- set a specific namespace, e.g. `kube-green-system`, as the namespace where to save kube-green. This could work, but by default operatorhub subscription set the namespace to `operators`, and it would result difficult to use for a final user.

Do you have some experiences about it, and could suggest me some other way to handle this?
Maybe the first possibility could be better, and if someone using olm if want to address other resources should install kube-green via helm chart?


I delved deeper into the issue, but it seems that with OLM there is no way to support the aggregator role with a cluster role (and cluster role binding). This limitation arises due to restrictions on the namespace name, and it seems there is no way to set a template for the namespace name.

Given this limitation, I could specify in the documentation that with OLM install method kube-green must be installed in the `operators` namespace (which is the default one recommended by OperatorHub). If you need to install in a different namespace, I would recommend using the Helm chart instead.

Other possibilities I considered include:

- Avoid using the aggregator role in OLM and grant kube-green permissions only on default managed resources. This would allow installation in any namespace, but the main feature of version 0.6 (handling all resource kinds) would become unusable
- Set a specific namespace, such as `kube-green-system`, as the designated namespace for kube-green. This could work, but by default, OperatorHub subscriptions set the namespace to operators, which might make it harder for end users.

Do you have any experience with this or suggestions for alternative approaches?
Maybe the first option is better: if someone using OLM wants to handle additional resources, they could install kube-green via the Helm chart instead. What do you think about?

**Comment by blezoray** [comment]
Force the namespace is really a bad solution. you said the default namespace for OperatorHub is `operators`, for Openshift, it is `openshift-operators`. And as a good pratice, we deploy each operator in a dedicated namespace to avoid any impact on the others operators.

I think the better solution is to remove the label `kube-green.dev/aggregate-to-manager`.

**Comment by davidebianchi** [comment]
Yes, I agree, all the suggested solutions are not ideal. This is why, before taking any action, I preferred to discuss what the best option was.

So, you’d prefer to have the ability to release kube-green in any namespace, even if that means only supporting the default resources and not adding support for custom ones. That works for me (it was actually the first alternative I thought of). If someone needs to support custom resources, it should install kube-green via helm chart until olm will add support to aggregator roles.

**Comment by blezoray** [comment]
I confirm. Today, our endusers expect to use kube-green with StatefulSet only.

**Comment by blezoray** [comment]
Hello, any update about this issue ? Do you plan to create a new version with the fix ?

**Comment by davidebianchi** [comment]
Hi @blezoray! I just created the 0.7.0 which brings various improvement to the configuration and it is included this fix.

**Comment by blezoray** [comment]
Thanks, I'll test it asap
